src/table/thead.tsx

```diff
@@ -176,6 +176,7 @@ const Thead = React.forwardRef(
                 // Expandable option is only applicable to the first data column of the table.
                 // When present, the header content receives extra padding to match the first offset in the data cells.
                 isExpandable={colIndex === 0 && isExpandable}
+                hasDynamicContent={column.hasDynamicContent}
```

Comment:
should we ignore it when sticky header is not enabled?

Reply:
yes, have now moved around the logic to make this clearer and more concise

Reply:
Makes sense, but should we still check for stickyHeader flag so that we don't attach observers if the feature is used in a table w/o sticky header, e.g. if the stickiness is conditional.

Reply:
`hidden` on [this line](https://github.com/cloudscape-design/components/pull/2842/files#diff-abe617070411a68e1443642bc6854265273a9e0d1452fc449b65a3d0a9355c8dR179) is only set if stickyHeader is true

Reply:
Ah, I see now.